### PR TITLE
feat: [PL-31798]: fix queries that fetches unlimited record for connectors and secrets

### PR DESCRIPTION
**Pull request by bhavya181**
## Describe your changes
fix queries that fetches unlimited record for connectors and secrets
Apply limit of 1000 to connectors listing API and limit of 50k to secrets listing API

tested locally by calling connector list APIs with size more than 1000, that throws error

Screenshot 2023-03-02 at 3 11 54 PM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44602)

### CODE REVIEW

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44602)* status: 3 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGSecretServiceV2Impl.java` line 390 at r2](https://reviewable.io/reviews/harness/harness-core/44602#-NPWvRvBC4SDcxgvN2_y:-NPWvRvBC4SDcxgvN2_z:b-hlfiin) ([raw file](https://github.com/harness/harness-core/blob/174c958f90b4713b10778637618e832e98ef88d5/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGSecretServiceV2Impl.java#L390)):*
> ```Java
>   @Override
>   public Page<Secret> list(Criteria criteria) {
>     return secretRepository.findAll(criteria, Pageable.ofSize(50000));
> ```

Do we need to perform check at service layer?
We can restrict at resource layer. I see there are 2 List APIs with page size parameter. Please have MaxSize check similar as done for connectors.

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44602)* status: 3 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGSecretServiceV2Impl.java` line 390 at r2](https://reviewable.io/reviews/harness/harness-core/44602#-NPWvRvBC4SDcxgvN2_y:-NPWxh-Q5JeHeR_APT9_:b-kgn51w) ([raw file](https://github.com/harness/harness-core/blob/174c958f90b4713b10778637618e832e98ef88d5/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGSecretServiceV2Impl.java#L390)):*
<details><summary><i>Previously, Jimit09Gandhi (Jimit Gandhi) wrote…</i></summary><blockquote>

Do we need to perform check at service layer?
We can restrict at resource layer. I see there are 2 List APIs with page size parameter. Please have MaxSize check similar as done for connectors.
</blockquote></details>

The reason I mention is because this the way we are performing RBAC check.
we will add limitation to no of secrets in account to 50k in List call. Although I think in real-world this case wont arise.

**Review by Jimit09Gandhi — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44602)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by Jimit09Gandhi — DISMISSED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44602#-:-NPXBCuxD3bqoU06anF-:bnfp4nl)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by ankit-harness — DISMISSED** [review]
> Reviewed 1 of 1 files at r4, 1 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44602)* status: 6 file(s) to review. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java` line 407 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPrGsh80P_YfdgKQyWJ:-NPrGsh80P_YfdgKQyWK:bcgjvre) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java#L407)):*
> ```Java
>     if (accessControlClient.hasAccess(ResourceScope.of(accountIdentifier, orgIdentifier, projectIdentifier),
>             Resource.of(SECRET_RESOURCE_TYPE, null), SECRET_VIEW_PERMISSION)) {
>       SortOrder order =
> ```

Why this change?

**Review by Jimit09Gandhi — DISMISSED** [review]
:lgtm:

> Reviewed 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44602#-:-NPrGz5A904RTge4gOdn:bnfp4nl)* status: 5 file(s) to review.

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -2756,6 +2756,7 @@ components:
       schema:
         type: integer
         default: 30
+        maximum: 1000
       description: 'Pagination: Number of items to return'
```
Comment (rashminandasahoo) [inline 120-ng-manager/contracts/openapi/v1/openapi.yaml:2760]:
```suggestion
      description: 'Number of items to return per page.'
```

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java [hunk 2]
```diff
@@ -300,8 +301,8 @@ public class NGSecretResourceV2 {
       @DefaultValue("false") boolean includeAllSecretsAccessibleAtScope,
       @Parameter(description = "Page number of navigation. The default value is 0") @QueryParam(
           NGResourceFilterConstants.PAGE_KEY) @DefaultValue("0") int page,
-      @Parameter(description = "Number of entries per page. The default value is 100 ") @QueryParam(
-          NGResourceFilterConstants.SIZE_KEY) @DefaultValue("100") int size) {
+      @Parameter(description = "Number of entries per page. The default value is 100. Max value is 1000 ") @QueryParam(
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java:304]:
```suggestion
      @Parameter(description = "Number of entries per page. The default number of entries per page is 100, while the maximum number allowed is 1000.") @QueryParam(
```

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java [hunk 3]
```diff
@@ -300,8 +301,8 @@ public class NGSecretResourceV2 {
       @DefaultValue("false") boolean includeAllSecretsAccessibleAtScope,
       @Parameter(description = "Page number of navigation. The default value is 0") @QueryParam(
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java:302]:
```suggestion
      @Parameter(description = "Navigation page number. By default, it is set to 0.") @QueryParam(
```

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java [hunk 4]
```diff
@@ -328,8 +329,8 @@ public class NGSecretResourceV2 {
       String projectIdentifier, @Body SecretResourceFilterDTO secretResourceFilterDTO,
       @Parameter(description = "Page number of navigation. The default value of 0") @QueryParam(
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java:330]:
```suggestion
      @Parameter(description = "Navigation page number. By default, it is set to 0.") @QueryParam(
```

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java [hunk 5]
```diff
@@ -328,8 +329,8 @@ public class NGSecretResourceV2 {
...
           NGResourceFilterConstants.PAGE_KEY) @DefaultValue("0") int page,
-      @Parameter(description = "Number of entries per page. The default value is 100") @QueryParam(
-          NGResourceFilterConstants.SIZE_KEY) @DefaultValue("100") int size) {
+      @Parameter(description = "Number of entries per page. The default value is 100. Max value is 1000.") @QueryParam(
```
Comment (rashminandasahoo) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java:332]:
```suggestion
      @Parameter(description = "Number of entries per page. The default number of entries per page is 100, while the maximum number allowed is 1000.") @QueryParam(
```

File: 440-connector-nextgen/contracts/openapi/v1/openapi.yaml [hunk 6]
```diff
@@ -2040,6 +2040,7 @@ components:
       schema:
         type: integer
         default: 30
+        maximum: 1000
       description: 'Pagination: Number of items to return'
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/contracts/openapi/v1/openapi.yaml:2044]:
```suggestion
      description: 'Number of items to return per page.'
```

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 7]
```diff
@@ -212,8 +214,8 @@ public ConnectorResource(@Named("connectorDecoratorService") ConnectorService co
   public ResponseDTO<PageResponse<ConnectorResponseDTO>>
   list(@Parameter(description = "Page number of navigation. The default value is 0") @QueryParam(
            NGResourceFilterConstants.PAGE_KEY) @DefaultValue("0") int page,
-      @Parameter(description = "Number of entries per page. The default value is 100") @QueryParam(
-          NGResourceFilterConstants.SIZE_KEY) @DefaultValue("100") int size,
+      @Parameter(description = "Number of entries per page. The default value is 100. Max value is 1000") @QueryParam(
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:217]: Same as suggested above.

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 8]
```diff
@@ -250,8 +252,8 @@ public ConnectorResource(@Named("connectorDecoratorService") ConnectorService co
   public ResponseDTO<PageResponse<ConnectorResponseDTO>>
   list(@Parameter(description = "Page number of navigation. The default value is 0") @QueryParam(
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:253]: Same as suggested above.

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 9]
```diff
@@ -250,8 +252,8 @@ public ConnectorResource(@Named("connectorDecoratorService") ConnectorService co
...
            NGResourceFilterConstants.PAGE_KEY) @DefaultValue("0") int page,
-      @Parameter(description = "Number of entries per page. The default value is 100") @QueryParam(
-          NGResourceFilterConstants.SIZE_KEY) @DefaultValue("100") int size,
+      @Parameter(description = "Number of entries per page. The default value is 100. Max value is 1000") @QueryParam(
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:255]: Same as suggested above.

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 10]
```diff
@@ -294,8 +296,8 @@ public ConnectorResource(@Named("connectorDecoratorService") ConnectorService co
   public ResponseDTO<PageResponse<CombineCcmK8sConnectorResponseDTO>>
   ccmK8sList(@Parameter(description = "Page number of navigation. The default value is 0") @QueryParam(
                  NGResourceFilterConstants.PAGE_KEY) @DefaultValue("0") int page,
-      @Parameter(description = "Number of entries per page. The default value is 100") @QueryParam(
-          NGResourceFilterConstants.SIZE_KEY) @DefaultValue("100") int size,
+      @Parameter(description = "Number of entries per page. The default value is 100. Max value is 1000") @QueryParam(
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:299]: Same as suggested above.

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 11]
```diff
@@ -556,8 +558,11 @@ public ConnectorResource(@Named("connectorDecoratorService") ConnectorService co
   public ResponseDTO<List<ConnectorResponseDTO>>
   listConnectorByFQN(@Parameter(description = ACCOUNT_PARAM_MESSAGE, required = true) @NotBlank @QueryParam(
                          NGCommonEntityConstants.ACCOUNT_KEY) String accountIdentifier,
-      @RequestBody(
-          required = true, description = "List of ConnectorsFQN as strings") @Body List<String> connectorsFQN) {
+      @RequestBody(required = true,
+          description = "List of ConnectorsFQN as strings. Max size is 1000") @Body List<String> connectorsFQN) {
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:562]:
```suggestion
          description = "A list of connectors' FQNs as strings. A maximum of 1000 characters is allowed.") @Body List<String> connectorsFQN) {
```

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 12]
```diff
@@ -556,8 +558,11 @@ public ConnectorResource(@Named("connectorDecoratorService") ConnectorService co
...
+    if (connectorsFQN.size() > MAX_LIMIT) {
+      throw new InvalidRequestException("List of ConnectorsFQN must be less than or equal to 1000");
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:564]:
```suggestion
      throw new InvalidRequestException("The FQNs of the connectors should be less than or equal to 1000.");
```

File: 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java [hunk 13]
```diff
@@ -633,6 +638,9 @@ public ResponseDTO<List<Map<String, String>>> getConnectorsAttributes(
       @QueryParam(NGCommonEntityConstants.ORG_KEY) String orgIdentifier,
       @QueryParam(NGCommonEntityConstants.PROJECT_KEY) String projectIdentifier,
       @QueryParam("connectorIdentifiers") List<String> connectorIdentifiers) {
+    if (connectorIdentifiers.size() > MAX_LIMIT) {
+      throw new InvalidRequestException("List of connectorIdentifiers must be less than or equal to 1000");
```
Comment (rashminandasahoo) [inline 440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java:642]:
```suggestion
      throw new InvalidRequestException("The number of connector identifiers should be less than or equal to 1000.");
```

**Review by rashminandasahoo — CHANGES_REQUESTED** [review]

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44602)* status: 6 file(s) to review. 1 change requests. 13. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java` line 407 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPrGsh80P_YfdgKQyWJ:-NQ4mAof0ecY_svhjxmZ:btrl95e) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java#L407)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Why this change?
</blockquote></details>

If user has permission to view secrets, then we can just fetch paginated secrets, there is no need to fetching all secrets and then applying pagination for that case.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java` line 302 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPX6pxmVdcOhjKaScrs-r5-302:-NQ4m2Gc86XYxNC7CDmF:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java#L302)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

```suggestion
      @Parameter(description = "Navigation page number. By default, it is set to 0.") @QueryParam(
```
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java` line 304 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPX6pxmVdcOhjKaScrs-r5-304:-NQ4lbFT2Bu5vsLq2fi9:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java#L304)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

```suggestion
      @Parameter(description = "Number of entries per page. The default number of entries per page is 100, while the maximum number allowed is 1000.") @QueryParam(
```
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java` line 330 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPX6pxmVdcOhjKaScrs-r5-330:-NQ4m38tCS4ASg-T-pi4:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java#L330)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

```suggestion
      @Parameter(description = "Navigation page number. By default, it is set to 0.") @QueryParam(
```
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java` line 332 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPX6pxmVdcOhjKaScrs-r5-332:-NQ4m3sfCKk79qOcPWjW:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/120-ng-manager/src/main/java/io/harness/ng/core/remote/NGSecretResourceV2.java#L332)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

```suggestion
      @Parameter(description = "Number of entries per page. The default number of entries per page is 100, while the maximum number allowed is 1000.") @QueryParam(
```
</blockquote></details>

Done.
___
*[`440-connector-nextgen/contracts/openapi/v1/openapi.yaml` line 2044 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPWguCGlPjdkTw1fbYp-r5-2044:-NQ4m4dK7mjnccysdAEz:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/440-connector-nextgen/contracts/openapi/v1/openapi.yaml#L2044)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

```suggestion
      description: 'Number of items to return per page.'
```
</blockquote></details>

Done.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java` line 217 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPWguCGlPjdkTw1fbYq-r5-217:-NQ4m5Fk9pAcMQJ90fJf:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java#L217)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Same as suggested above.
</blockquote></details>

Done.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java` line 253 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPWguCGlPjdkTw1fbYq-r5-253:-NQ4m682-tazMsIaU4Xe:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java#L253)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Same as suggested above.
</blockquote></details>

Done.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java` line 255 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPWguCGlPjdkTw1fbYq-r5-255:-NQ4m6aw1PO6o8y0ov24:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java#L255)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Same as suggested above.
</blockquote></details>

Done.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java` line 299 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPWguCGlPjdkTw1fbYq-r5-299:-NQ4m7Ml0s9I20rhSlxo:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java#L299)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Same as suggested above.
</blockquote></details>

Done.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java` line 562 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPWguCGlPjdkTw1fbYq-r5-562:-NQ4m7v_2uJLUB8RzFfk:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java#L562)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

```suggestion
          description = "A list of connectors' FQNs as strings. A maximum of 1000 characters is allowed.") @Body List<String> connectorsFQN) {
```
</blockquote></details>

Done.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java` line 564 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPWguCGlPjdkTw1fbYq-r5-564:-NQ4m8MMApXm4RczAN5u:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java#L564)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

```suggestion
      throw new InvalidRequestException("The FQNs of the connectors should be less than or equal to 1000.");
```
</blockquote></details>

Done.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java` line 642 at r5](https://reviewable.io/reviews/harness/harness-core/44602#-NPWguCGlPjdkTw1fbYq-r5-642:-NQ4m8lS2W_hHLrxL307:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7a51b3e70fdbe76007196d41fbaeb5dd77b763e9/440-connector-nextgen/src/main/java/io/harness/connector/apis/resource/ConnectorResource.java#L642)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

```suggestion
      throw new InvalidRequestException("The number of connector identifiers should be less than or equal to 1000.");
```
</blockquote></details>

Done.

**Review by rashminandasahoo — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44602)* status: 6 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by ankit-harness — DISMISSED** [review]
:lgtm:

> Reviewed 2 of 3 files at r5, 3 of 4 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44602#-:-NQ4teAN1NnJ-3yTxkAE:bnfp4nl)* status: 4 file(s) to review.

**Review by Jimit09Gandhi — DISMISSED** [review]
:lgtm:

> Reviewed 3 of 4 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44602#-:-NQ4vGcg1pq8iHYrb9Pk:bnfp4nl)* status: :shipit: complete! ✓

**Review by rashminandasahoo — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44602#-:-NQ5MwGf7Rllcg8t-MoV:bnfp4nl)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Pull request merged**